arch/arm/core/aarch32/cortex_m/fpu.c

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2019,2020 Linaro Limited
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <arch/arm/aarch32/cortex_m/cmsis.h>
+
+/**
+ * @file @brief Helper functions for save and restoring the FP context
+ *
+ */
+
+int save_fp_context(uint32_t *buffer, size_t buffer_size_words, bool *context_saved)
```

Comment:
Since the functions are now part of the ARCH/ARM internal API, then need to have the `z_arm_` prefix, as most of the exposed internal API